### PR TITLE
Capability exchanges

### PR DESCRIPTION
**Pull request by ewuerger**
This PR increases coverage of endpoints of the SystemArchitectureLayer. Specifically the Mission receives new attributes: exploits, involvements and exploitations. `.exploits` works as ashortcut to get to the exploited OperationalCapabilities rightaway. Capability and OperationalCapability Exchanges: include, extend and generalizations are added. Special focus on `.incoming_exploitations` which is a mandatory attribute on Capabilites enabling fast tracking of which Missions are in the context of a Capability.

Then for rendering of [MCB] and [OCB] diagrams it was needed to add default styling. Additionally we finally did a refactoring of svg.drawing.py where `LabelBuilder` and `LinesData` were added to remove the amount of parameters. These were similar on various methods. On top we refactored the derimental _draw_label method such that it is more readable and critical steps are outsourced in their own function. This is very needed b/c _draw_label is the core label producing method called in box, edge and symbol labeling.

Finally the text in OperationalCapabilitySymbol (OC) was way too bold and didn't reuse _brown_oval that is used by mission and capability symbol factories. This was also changed.

### CODE REVIEW

**Review by Wuestengecko — CHANGES_REQUESTED** [review]
I'm gonna make the bold assumption that you have tested all the new calculations that you introduced, so I only have a few minor points about code style to make. ;)

File: capellambse/model/layers/ctx.py [hunk 1]
```diff
@@ -17,6 +17,7 @@
 
 XT_ARCH = "org.polarsys.capella.core.data.ctx:SystemAnalysis"
 XT_CAP_INV = "org.polarsys.capella.core.data.ctx:CapabilityInvolvement"
+XT_CAP_EXP = "org.polarsys.capella.core.data.ctx:CapabilityExploitation"
```
[the remark this answers is not in the document]
Comment (ewuerger) [reply capellambse/model/layers/ctx.py:20]: You are right. I can swap the string with the classtype for xtypes in [Mission.exploits](https://github.com/DSD-DBS/py-capellambse/pull/102/files/9dff0998a5d6e4d193c19e4f83899a777d636675#diff-9461c9e412f391a413ece50fb2a1d724aeecba17a069cb5c44b79cdbf898358bR171-R176). Very good catch!

File: capellambse/svg/drawing.py [hunk 2]
```diff
@@ -195,46 +214,30 @@ def _draw_feature_text(
...
+            debug_y = int(builder.label["y"]) + y_margin
             debug_y1 = (
-                int(label["y"])
-                + (int(label["height"]) - decorations.icon_size) / 2
+                int(builder.label["y"])
+                + (int(builder.label["height"]) - decorations.icon_size) / 2
             )
             x = (
-                int(label["x"])
+                int(builder.label["x"])
```
[the remark this answers is not in the document]
Comment (ewuerger) [reply capellambse/svg/drawing.py:240]: I don't know. Even some lines after that `build.label["width']` is used without casting. Therefore changed as requested.

File: capellambse/svg/generate.py [hunk 3]
```diff
@@ -108,7 +108,11 @@ def from_json_path(cls, path: str | os.PathLike) -> SVGDiagram:
     def draw_object(self, obj: ContentsDict) -> None:
         self.drawing.draw_object(obj)
 
-    def save_drawing(self, pretty: bool = False, indent: int = 2) -> None:
+    def save_drawing(
+        self, pretty: bool = False, indent: int = 2, filename: str = ""
+    ) -> None:
+        if filename:
+            self.drawing.filename = filename
```
[the remark this answers is not in the document]
Comment (ewuerger) [reply capellambse/svg/generate.py]: That's why I like you to review my code. Very nice thank you.

File: capellambse/model/layers/ctx.py [hunk 4]
```diff
@@ -247,6 +291,17 @@ class SystemAnalysis(crosslayer.BaseArchitectureLayer):
...
+c.set_accessor(
+    Capability,
+    "incoming_exploitations",
+    c.CustomAccessor(
+        CapabilityExploitation,
+        operator.attrgetter("_model.sa.all_capability_exploitations"),
+        elmmatcher=operator.eq,
+        matchtransform=operator.attrgetter("capability"),
+        aslist=c.ElementList,
+    ),
```
[the remark this answers is not in the document]
Comment (ewuerger) [reply capellambse/model/layers/ctx.py:301]: +1

**Review by Wuestengecko — APPROVED** [review]
Looks good now. LGTM.

**Pull request merged**